backend/apps/ifc_validation_bff/views_legacy.py

```diff
@@ -100,50 +100,48 @@ def file_contains_string(file_name, fragment):
 
 @functools.lru_cache(maxsize=1024)
 def get_feature_url(feature_code, feature_version):
+    # TODO - fetch remote/sha based on code/version; for now we point to main and development repo (configurable)
+    file_folder = os.path.dirname(os.path.realpath(__file__))
+    rules_folder = os.path.join(file_folder, '../ifc_validation/checks/ifc_gherkin_rules/features/rules')
 
-    # TODO - fetch remote/sha based on code/version; for now we point to main repo (configurable)
+    version_tag = f'@version{feature_version}'
 
-    file_folder = os.path.dirname(os.path.realpath(__file__))
-    feature_folder = os.path.join(file_folder, '../ifc_validation/checks/ifc_gherkin_rules/features')
+    for root, _, files in os.walk(rules_folder):
+        for file in files:
+            file_name = os.fsdecode(file)
+            fq_file_name = os.path.join(root, file_name)
 
-    for file in os.listdir(feature_folder):
-        file_name = os.fsdecode(file)
-        fq_file_name = os.path.join(feature_folder, file_name)
-        version_tag = f'@version{feature_version}'
-        
-        if file_name.endswith(".feature") and file_name.startswith(feature_code) and file_contains_string(fq_file_name, version_tag):
-            return FEATURE_URL + file_name
+            if file_name.endswith(".feature") and file_name.startswith(feature_code) and file_contains_string(fq_file_name, version_tag):
+                return os.path.join(FEATURE_URL, feature_code[:3] , file_name) 
```

Comment:
I don't really understand the intended behaviour here.

We read the file contents for the version and if it matches we return stuff.

But what the file doesn't match the version?

- I think we should at least break out of the loop. There is no reason to keep scanning for file name matches there is only 1 feature file that will match the pattern.
- But if we don't go back in time, why would we consider the version? Isn't it better to just return something even if it's the wrong version?
- I'd probably adapt this to use `glob` to just construct the pattern, e.g `"./ALB/ALB001*.feature"`. Easier to read, adapt, and break out of.